src/sentry/eventstream/snuba.py

```diff
@@ -1,15 +1,251 @@
+        self._send(
+            state['project_id'],
+            'end_delete_groups',
+            extra_data=(state,),
+            asynchronous=False
+        )
+
+    def start_merge(self, project_id, previous_group_ids, new_group_id):
+        if not previous_group_ids:
+            return
+
+        state = {
+            'transaction_id': uuid4().hex,
+            'project_id': project_id,
+            'previous_group_ids': list(previous_group_ids),
+            'new_group_id': new_group_id,
+            'datetime': datetime.now(tz=pytz.utc),
+        }
+
+        self._send(
+            project_id,
+            'start_merge',
+            extra_data=(state,),
+            asynchronous=False
+        )
+
+        return state
+
+    def end_merge(self, state):
+        state = state.copy()
+        state['datetime'] = datetime.now(tz=pytz.utc)
+        self._send(
+            state['project_id'],
+            'end_merge',
+            extra_data=(state,),
+            asynchronous=False
+        )
+
+    def start_unmerge(self, project_id, hashes, previous_group_id, new_group_id):
+        if not hashes:
+            return
+
+        state = {
+            'transaction_id': uuid4().hex,
+            'project_id': project_id,
+            'previous_group_id': previous_group_id,
+            'new_group_id': new_group_id,
+            'hashes': list(hashes),
+            'datetime': datetime.now(tz=pytz.utc),
+        }
+
+        self._send(
+            project_id,
+            'start_unmerge',
+            extra_data=(state,),
+            asynchronous=False
+        )
+
+        return state
+
+    def end_unmerge(self, state):
+        state = state.copy()
+        state['datetime'] = datetime.now(tz=pytz.utc)
+        self._send(
+            state['project_id'],
+            'end_unmerge',
+            extra_data=(state,),
+            asynchronous=False
+        )
+
+    def start_delete_tag(self, project_id, tag):
+        if not tag:
+            return
+
+        state = {
+            'transaction_id': uuid4().hex,
+            'project_id': project_id,
+            'tag': tag,
+            'datetime': datetime.now(tz=pytz.utc),
+        }
+
+        self._send(
+            project_id,
+            'start_delete_tag',
+            extra_data=(state,),
+            asynchronous=False
+        )
+
+        return state
+
+    def end_delete_tag(self, state):
+        state = state.copy()
+        state['datetime'] = datetime.now(tz=pytz.utc)
+        self._send(
+            state['project_id'],
+            'end_delete_tag',
+            extra_data=(state,),
+            asynchronous=False
+        )
+
+    def _send(self, project_id, _type, extra_data=(), asynchronous=True):
+        raise NotImplementedError
 
 
-class SnubaEventStream(KafkaEventStream):
+class SnubaEventStream(SnubaProtocolEventStream):
     def _send(self, project_id, _type, extra_data=(), asynchronous=True):
```

Comment:
It sort of seems like this should default to `False` and assert that it's not `True` (since this backend doesn't support asynchronous operations), but that would contradict the behavior of `KafkaEventStream` and I'm not sure it's worth making a fuss over.